src/scene/renderer/forward-renderer.js

```diff
@@ -768,9 +772,12 @@ class ForwardRenderer {
 
                 shadowCamView.setTRS(cookieNode.getPosition(), cookieNode.getRotation(), Vec3.ONE).invert();
                 shadowCamViewProj.mul2(cookieCam.projectionMatrix, shadowCamView);
-                spot._shadowMatrix.mul2(ShadowRenderer.scaleShiftMatrix, shadowCamViewProj);
+
+                if (!spot._cookieMatrix) spot._cookieMatrix = new Mat4();
```

Comment:
Could you just hide this in the `Light` class? So in the getter of `Light#cookieMatrix` check if it's `null` and create it if necessary?

Reply:
absolutely! done.